src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,172 @@
+#ifndef PHARE_SPLIT_2D_H
+#define PHARE_SPLIT_2D_H
+namespace PHARE::amr
+{
+template<> // Generic 2D delta setter
+struct SplitInnerSetter</*dim=*/2, 0>
+{
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal& delta_vals, size_t root)
+    {
+        for (size_t y = 0; y < root; y++)
+            for (size_t yoff = y * root, x = 0; x < root; x++)
+                deltas[x + yoff] = {delta_vals[x], delta_vals[y]};
+    }
+};
+
+
+/*************************************************************************
+  dim = 2
+  interp = 1
+  nbrRefineParticles = 9
+*/
+template<>
+class Splitter<2, 1, 9> : public ASplitter<2, 1, 9>
+{
+public:
+    Splitter()
+        : ASplitter{weight_vals_, delta_vals_}
+    {
+    }
+
+protected:
+    constexpr static float delta_vals_[]  = {-1, 0, 1};
+    constexpr static float weight_vals_[] = {0.25, 0.125, 0.01625};
+};
+
+template<>
+struct SplitInnerSetter</*dim=*/2, /*nbrRefineParticles*/ 9>
+{
+    constexpr static size_t dim  = 2;
+    constexpr static size_t root = 3; // 3*3
```

Comment:
I don't think this strategy will work for other numbers that are not perfect square numbers...
like for dim=2, N=5, or dim=2, N=8

Reply:
this is a function for a perfect square, each combination of N will require it's own layout function

Reply:
or possibly the square can be an aggregate of smaller Ns